modules/data-streams/src/javaRestTest/java/org/elasticsearch/datastreams/logsdb/qa/StandardVersusLogsIndexModeChallengeRestIT.java

```diff
@@ -0,0 +1,278 @@
+
+        Matcher.mappings(getContenderMappings(), getBaselineMappings())
+            .settings(getContenderSettings(), getBaselineSettings())
+            .expected(getResponseSourceAsMap(queryBaseline(searchSourceBuilder)))
+            .ignoringSort(true)
+            .isEqualTo(getResponseSourceAsMap(queryContender(searchSourceBuilder)));
+    }
+
+    public void testHistogramAggregation() throws IOException, MatcherException {
+        final List<XContentBuilder> documents = new ArrayList<>();
+        int numberOfDocuments = randomIntBetween(100, 200);
+        for (int i = 0; i < numberOfDocuments; i++) {
+            documents.add(generateDocument(Instant.now().plus(i, ChronoUnit.SECONDS)));
+        }
+
+        final Tuple<Response, Response> tuple = indexDocuments(() -> documents, () -> documents);
+        assertThat(tuple.v1().getStatusLine().getStatusCode(), Matchers.equalTo(RestStatus.OK.getStatus()));
+        assertThat(tuple.v2().getStatusLine().getStatusCode(), Matchers.equalTo(RestStatus.OK.getStatus()));
+
+        final SearchSourceBuilder searchSourceBuilder = new SearchSourceBuilder().query(QueryBuilders.matchAllQuery())
+            .size(numberOfDocuments)
+            .aggregation(new HistogramAggregationBuilder("agg").field("memory_usage_bytes").interval(100.0D));
+
+        Matcher.mappings(getContenderMappings(), getBaselineMappings())
+            .settings(getContenderSettings(), getBaselineSettings())
+            .expected(getResponseSourceAsMap(queryBaseline(searchSourceBuilder)))
+            .ignoringSort(true)
+            .isEqualTo(getResponseSourceAsMap(queryContender(searchSourceBuilder)));
+    }
+
+    public void testTermsAggregation() throws IOException, MatcherException {
+        final List<XContentBuilder> documents = new ArrayList<>();
+        int numberOfDocuments = randomIntBetween(100, 200);
+        for (int i = 0; i < numberOfDocuments; i++) {
+            documents.add(generateDocument(Instant.now().plus(i, ChronoUnit.SECONDS)));
+        }
+
+        final Tuple<Response, Response> tuple = indexDocuments(() -> documents, () -> documents);
+        assertThat(tuple.v1().getStatusLine().getStatusCode(), Matchers.equalTo(RestStatus.OK.getStatus()));
+        assertThat(tuple.v2().getStatusLine().getStatusCode(), Matchers.equalTo(RestStatus.OK.getStatus()));
+
+        final SearchSourceBuilder searchSourceBuilder = new SearchSourceBuilder().query(QueryBuilders.matchAllQuery())
+            .size(numberOfDocuments)
+            .size(0)
+            .aggregation(new TermsAggregationBuilder("agg").field("host.name"));
+
+        Matcher.mappings(getContenderMappings(), getBaselineMappings())
+            .settings(getContenderSettings(), getBaselineSettings())
+            .expected(getResponseSourceAsMap(queryBaseline(searchSourceBuilder)))
+            .ignoringSort(true)
+            .isEqualTo(getResponseSourceAsMap(queryContender(searchSourceBuilder)));
```

Comment:
Are we doing any validation in this case? There is no hits in results of this query.

Reply:
I will add it

Reply:
So this is an aggregation...and I do `setSize(0)` to just return aggregations and no document hit...as a result it is correct that there is no hit.

Reply:
I see what you mean...for an aggregation we need to compare buckets not hits... 🤦